src/net/sourceforge/kolmafia/session/ChibiBuddyManager.java

```diff
@@ -19,15 +23,87 @@ public class ChibiBuddyManager {
 
   private ChibiBuddyManager() {}
 
+  private static String randomName() {
+    var contacts =
+        ContactManager.getMailContacts().stream().filter(Predicate.not(String::isEmpty)).toList();
+    var rng = new Random();
+    var player =
+        (contacts.size() == 0)
+            ? KoLCharacter.getUserName()
+            : contacts.get(rng.nextInt(contacts.size()));
+    return "Li'l " + player;
+  }
+
+  public static void ensureLiveChibi() {
+    if (haveChibiBuddyOn()) {
+      RequestThread.postRequest(
+          new GenericRequest("inv_use.php?whichitem=" + ItemPool.CHIBIBUDDY_ON));
+      RequestThread.postRequest(new GenericRequest("choice.php?whichchoice=627&option=7"));
+    }
+
+    if (!haveChibiBuddyOn()) {
+      var req = new GenericRequest("inv_use.php?whichitem=" + ItemPool.CHIBIBUDDY_OFF);
+      RequestThread.postRequest(req);
+      if (ChoiceManager.currentChoice() == 633) {
+        // Users can provide their own names using the choice adventure prefs.
```

Comment:
does this involve setting it to `1&chibiname=NAME`? If so, does that mean that `choiceAdventure633` is the first which isn't necessarily an integer? Does this interface well with everything else?

Reply:
It's not the first, this is a thing across mafia already! See ChoiceManager.automateChoice for example